### PR TITLE
Expose non persisted identifier for message

### PR DESCRIPTION
**Pull request by alexaubry**
### Issues

In UI, we are using the nonce of the message to cache the cell descriptions. This causes a problem when the message is edited, because the nonce changes, which causes race conditions when the edit notification are sent.

### Solutions

- We expose `nonpersistedObjectIdentifer` on the message type protocol to use it in the cache
- We add a test to verify that this identifier does not change on edit

### CODE REVIEW

File: Source/Public/ZMManagedObject.h [hunk 1]
```diff
@@ -19,6 +19,8 @@
 
 @import CoreData;
 
+NS_ASSUME_NONNULL_BEGIN
```
Comment (typfel) [inline Source/Public/ZMManagedObject.h]: You might need do major release to catch any usage of this in other frameworks.

Comment (alexaubry) [reply Source/Public/ZMManagedObject.h]: Ah good point. Should I revert these changes then?

Comment (alexaubry) [reply Source/Public/ZMManagedObject.h]: @typfel Reverted. Can you have a look again?

**Comment by alexaubry** [comment]
@zenkins release minor

**Pull request merged**